optimum/intel/openvino/modeling_text2speech.py

```diff
@@ -0,0 +1,678 @@
+        self._device = device.upper()
+        self.is_dynamic = dynamic_shapes
+        self.ov_config = {} if ov_config is None else {**ov_config}
+        self.preprocessors = kwargs.get("preprocessors", [])
+
+        self.encoder_model = encoder_model
+        self.decoder_model = decoder_model
+        self.postnet_model = postnet_model
+        self.vocoder_model = vocoder_model
+
+        self._supports_cache_class = False
+        self.main_input_name = "input_ids"
+        self._compile_only = kwargs.get("compile_only", False)
+
+        enable_compilation = kwargs.get("compile", True)
+        self.generation_config = kwargs.get("generation_config", GenerationConfig.from_model_config(config))
+        self._openvino_config = None
+        if quantization_config:
+            self._openvino_config = OVConfig(quantization_config=quantization_config)
+        self._set_ov_config_parameters()
+        self.encoder_model = OVTextToSpeechEncoder(self.encoder_model, self)
+        self.decoder_model = OVTextToSpeechDecoder(self.decoder_model, self)
+        self.postnet_model = OVTextToSpeechPostNet(self.postnet_model, self)
+        self.vocoder_model = OVTextToSpeechVocoder(self.vocoder_model, self)
+
+        if enable_compilation and not self._compile_only:
+            self.compile()
+
+        # Avoid warnings when creating a transformers pipeline
+        AutoConfig.register(self.base_model_prefix, AutoConfig)
+        try:
+            self.auto_model_class.register(AutoConfig, self.__class__)
+        except AttributeError:
+            pass
+
+    def clear_requests(self):
+        if self._compile_only:
+            raise ValueError(
+                "`clear_requests()` is not supported with `compile_only` mode, please intialize model without this option"
+            )
+
+        for _, component in self.components.items():
+            component.clear_requests()
+
+    def compile(self):
+        for _, component in self.components.items():
+            if isinstance(component, OVModelPart):
+                component._compile()
+            else:
+                component.compile()
+
+    @property
+    def _component_names(self):
+        component_names = ["encoder_model", "decoder_model", "postnet_model", "vocoder_model"]
+        return component_names
+
+    @property
+    def components(self):
+        return {component_name: getattr(self, component_name) for component_name in self._component_names}
+
+    @property
+    def _submodel_names(self):
+        model_names = ["encoder_model", "decoder_model", "postnet_model", "vocoder_model"]
+        return model_names
+
+    @property
+    def submodels(self):
+        return {submodel_name: getattr(self, submodel_name) for submodel_name in self._submodel_names}
+
+    @property
+    def ov_submodels(self) -> Dict[str, ov.runtime.Model]:
+        return {submodel_name: getattr(self, submodel_name).model for submodel_name in self._submodel_names}
```

Comment:
we should only define properties we need, components and submodels are the same thing here, we only defined components for diffusion pipelines because `DiffusionPipeline` class needs that property.

Reply:
Removed unneeded submodule properties and leave only components that are use in `_save_pretrained` method